packages/vscode-extension/src/builders/buildAndroid.ts

```diff
@@ -65,12 +66,15 @@ export async function getAndroidBuildPaths(
   return { apkPath, packageName };
 }
 
-function makeBuildTaskName(productFlavor: string, buildType: string) {
-  // task name is in the format of assemble<ProductFlavor><BuildType> where productFlavor and buildType
+function makeBuildTaskName(productFlavor: string, buildType: string, appName?: string) {
+  // task name is in the format of :<appName>:assemble<ProductFlavor><BuildType> where productFlavor and buildType
   // are the names of the productFlavor and buildType that each start with a capital letter.
   // By default, Android Gradle Plugin always creates staging and release buildTypes and does not define any productFlavor.
+  // and appName is the name of the application as seen by the gradle build system, omitting appName leads to
+  // whole android file being build.
   const flavor = productFlavor.charAt(0).toUpperCase() + productFlavor.slice(1);
-  return "assemble" + flavor + buildType.charAt(0).toUpperCase() + buildType.slice(1);
+  const prefix = appName ? `:${appName}:` : "";
+  return prefix + "assemble" + flavor + buildType.charAt(0).toUpperCase() + buildType.slice(1);
```

Comment:
```suggestion
  const flavorUppercase = productFlavor && productFlavor.charAt(0).toUpperCase() + productFlavor.slice(1);
  const buildTypeUppercase = buildType && buildType.charAt(0).toUpperCase() + buildType.slice(1);
  const prefix = appName ? `:${appName}:` : "";
  return `${prefix}assemble${flavorUppercase}${buildTypeUppercase}`;
```
1. check for emptiness, or assert non-emptiness elsewhere
1. store the build type in an intermediate variable to make the return statement clearer
1. I find format strings nicer to read, but that's a stylistic choice